astropy/visualization/wcsaxes/coordinate_helpers.py

```diff
@@ -186,10 +186,12 @@ def grid(self, draw_grid=True, grid_type=None, **kwargs):
 
         self.grid_lines_kwargs.update(kwargs)
 
-        if self.grid_lines_kwargs["visible"]:
-            if not draw_grid:
-                self.grid_lines_kwargs["visible"] = False
-        else:
+        if draw_grid:
+            self.grid_lines_kwargs["visible"] = True
+        elif draw_grid is False:
+            self.grid_lines_kwargs["visible"] = False
+        elif draw_grid is None:
+            draw_grid = True
```

Comment:
The variable isn't used after this conditional, so I think we don't need to rebind it
```suggestion
```